python_modules/dagster-graphql/dagster_graphql/schema/solids.py

```diff
@@ -385,10 +386,18 @@ def __init__(self, represented_pipeline: RepresentedJob, solid_def_name: str):
         self._solid_def_snap = represented_pipeline.get_node_def_snap(solid_def_name)
 
     def resolve_metadata(self, _graphene_info):
-        return [
-            GrapheneMetadataItemDefinition(key=item[0], value=item[1])
-            for item in self._solid_def_snap.tags.items()
-        ]
+        metadata_items = []
+        for key, val in self._solid_def_snap.tags.items():
+            metadata_items.append(GrapheneMetadataItemDefinition(key=key, value=val))
+            # Backcompat for legacy system tags. Older code servers may report the compute kind under
+            # the legacy tag. Code servers running versions after deprecation of the legacy tag may
+            # have both the legacy and current tag set. We can't patch this at the Snap level
+            # because the host process will complain about mismatched snap IDs
+            if key == LEGACY_COMPUTE_KIND_TAG and COMPUTE_KIND_TAG not in self._solid_def_snap.tags:
+                metadata_items.append(
+                    GrapheneMetadataItemDefinition(key=COMPUTE_KIND_TAG, value=val)
+                )
```

Comment:
should this if / else ?

Reply:
hmm-- following the same strategy here that I was in the snap patch, which is to just add the new tag instead of overwriting the old one, since users might have manually provided the old one and would be confused if it disappears. So I think no.